src/pages/VideoListing/VideoListing.jsx

```diff
@@ -0,0 +1,26 @@
+import "./video-listing.css";
+import { useEffect } from "react";
+import { useFetch } from "../../hooks/useFetch";
+import { Loader, VideoCard, Error } from "../../components";
+import { useVideos } from "../../context";
+import { SET_VIDEOS } from "../../helper/constants";
+
+const VideoListing = () => {
+  const { videoState, videoDispatch } = useVideos();
+  const { allVideos } = videoState;
+  const { loader, videoData, error } = useFetch("/api/videos");
+  useEffect(() => {
+    videoDispatch({ type: SET_VIDEOS, payload: videoData });
+  }, [videoData, videoDispatch]);
+
+  return (
+    <div className={`flex video__container {showMobileNav? "no-scroll":""}`}>
+      {loader && <Loader />}
+      {allVideos !== [] &&
```

Comment:
better way could be to check allVideos's length like allVideos.length !== 0

Reply:
Thankyou for your comments. I have made the suggested changes